cookbooks/bach_spark/templates/default/spark-defaults.conf.erb

```diff
@@ -0,0 +1,14 @@
+spark.driver.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraJavaOptions -verbose:gc -XX:+PrintHeapAtGC -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+UseNUMA -XX:+PrintGCApplicationStoppedTime -XX:+UseCompressedOops -XX:+PrintClassHistogram -XX:+PrintGCApplicationConcurrentTime
+spark.shuffle.consolidateFiles true
+spark.shuffle.io.numConnectionsPerPeer 2
+spark.history.fs.logDirectory <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.dir <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.enabled true
```

Comment:
Even eventLog is set, I still hit java.io.FileNotFoundException: File file:/tmp/spark-events does not exist Error. This results in SparkContext did not initialize after waiting for *****ms.